### DISCUSSION TITLE
Do longhorn volumes make sensible data available to untrustable containers?

### DISCUSSION THREAD

**Discussion by schlichtanders** [longhorn/longhorn#6796] [Q&A]
Hi there,
I am thinking a lot about how to secure containers from one another on the same cluster. I think I learned that the normal overlayFS, which empowers the running containers themselves with a writable layer, does not impose a leakage of data, because the underlying block storage cannot be accessed via the overlayFS.

With Longhorn however this seems different. Longhorn seems to create block storage from the node's default disk. Hence, it sounds like a longhorn volumes could make some block with hidden sensible information available to a potentially malicious other container which also has the respective apis to read out the entire block.

Is this reasoning true? If so, what is the recommended approach to prevent information leakage with longhorn?
If not, what makes longhorn secure?

**Comment by c3y1huang** [comment]
Here are some resources that should address your security concerns:
- [Volume Encryption](https://longhorn.io/docs/1.5.1/advanced-resources/security/volume-encryption/)
- [MTLS Support](https://longhorn.io/docs/1.5.1/advanced-resources/security/mtls-support/)

**Reply by schlichtanders** [reply]
Thank you for the pointers how to encrypt longhorn in rest and in transit.

Actually my nodes are already encrypted using LUKS. I could do a second encryption over the first encryption, but if Longhorn itself somehow by design solves this information-leakage-via-block-storage, I would prefer not to add a second encryption-in-rest layer.

**Reply by schlichtanders** [reply]
@c3y1huang can you also help on this follow up with double encryption?
